In [ ]:
from google.colab import drive
import pandas as pd
import seaborn as sns

In [ ]:
drive.mount('/content/gdrive')

Drive already mounted at /content/gdrive; to attempt to forcibly remount, call drive.mount("/content/gdrive", force_remount=True).


In [ ]:
dframe = pd.read_csv('gdrive/My Drive/MBAN_6110_Group_Report/heart_2022_no_nans.csv')
dframe.head()

State     Sex GeneralHealth  PhysicalHealthDays  MentalHealthDays  \
0  Alabama  Female     Very good                 4.0               0.0   
1  Alabama    Male     Very good                 0.0               0.0   
2  Alabama    Male     Very good                 0.0               0.0   
3  Alabama  Female          Fair                 5.0               0.0   
4  Alabama  Female          Good                 3.0              15.0   

                                     LastCheckupTime PhysicalActivities  \
0  Within past year (anytime less than 12 months ...                Yes   
1  Within past year (anytime less than 12 months ...                Yes   
2  Within past year (anytime less than 12 months ...                 No   
3  Within past year (anytime less than 12 months ...                Yes   
4  Within past year (anytime less than 12 months ...                Yes   

   SleepHours            RemovedTeeth HadHeartAttack  ... HeightInMeters  \
0         9.0            None of them             No  ...           1.60   
1         6.0            None of them             No  ...           1.78   
2         8.0  6 or more, but not all             No  ...           1.85   
3         9.0            None of them             No  ...           1.70   
4         5.0                  1 to 5             No  ...           1.55   

  WeightInKilograms    BMI AlcoholDrinkers HIVTesting FluVaxLast12  \
0             71.67  27.99              No         No          Yes   
1             95.25  30.13              No         No          Yes   
2            108.86  31.66             Yes         No           No   
3             90.72  31.32              No         No          Yes   
4             79.38  33.07              No         No          Yes   

  PneumoVaxEver                                  TetanusLast10Tdap  \
0           Yes                                 Yes, received Tdap   
1           Yes  Yes, received tetanus shot but not sure what type   
2           Yes  No, did not receive any tetanus shot in the pa...   
3           Yes  No, did not receive any tetanus shot in the pa...   
4           Yes  No, did not receive any tetanus shot in the pa...   

  HighRiskLastYear CovidPos  
0               No       No  
1               No       No  
2               No      Yes  
3               No      Yes  
4               No       No  

[5 rows x 40 columns]

In [ ]:
dframe.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 246022 entries, 0 to 246021
Data columns (total 40 columns):
 #   Column                     Non-Null Count   Dtype  
---  ------                     --------------   -----  
 0   State                      246022 non-null  object 
 1   Sex                        246022 non-null  object 
 2   GeneralHealth              246022 non-null  object 
 3   PhysicalHealthDays         246022 non-null  float64
 4   MentalHealthDays           246022 non-null  float64
 5   LastCheckupTime            246022 non-null  object 
 6   PhysicalActivities         246022 non-null  object 
 7   SleepHours                 246022 non-null  float64
 8   RemovedTeeth               246022 non-null  object 
 9   HadHeartAttack             246022 non-null  object 
 10  HadAngina                  246022 non-null  object 
 11  HadStroke                  246022 non-null  object 
 12  HadAsthma                  246022 non-null  object 
 13  HadSkinCancer              24

In [ ]:
df = dframe.copy()

In [ ]:
# check that all two-value columns have no outliers

print(df['Sex'].value_counts(dropna=False))
print()
print(df['PhysicalActivities'].value_counts(dropna=False))
print()
print(df['HadHeartAttack'].value_counts(dropna=False))
print()
print(df['HadAngina'].value_counts(dropna=False))
print()
print(df['HadStroke'].value_counts(dropna=False))
print()
print(df['HadAsthma'].value_counts(dropna=False))
print()
print(df['HadSkinCancer'].value_counts(dropna=False))
print()
print(df['HadCOPD'].value_counts(dropna=False))
print()
print(df['HadDepressiveDisorder'].value_counts(dropna=False))
print()
print(df['HadKidneyDisease'].value_counts(dropna=False))
print()
print(df['HadArthritis'].value_counts(dropna=False))
print()
print(df['DeafOrHardOfHearing'].value_counts(dropna=False))
print()
print(df['BlindOrVisionDifficulty'].value_counts(dropna=False))
print()
print(df['DifficultyConcentrating'].value_counts(dropna=False))
print()
print(df['DifficultyWalking'].value_counts(dropna=False))
print()
print(df['DifficultyDressingBathing'].value_counts(dropna=False))
print()
print(df['DifficultyErrands'].value_counts(dropna=False))
print()
print(df['ChestScan'].value_counts(dropna=False))
print()
print(df['AlcoholDrinkers'].value_counts(dropna=False))
print()
print(df['HIVTesting'].value_counts(dropna=False))
print()
print(df['FluVaxLast12'].value_counts(dropna=False))
print()
print(df['PneumoVaxEver'].value_counts(dropna=False))
print()
print(df['HighRiskLastYear'].value_counts(dropna=False))

Sex
Female    127811
Male      118211
Name: count, dtype: int64

PhysicalActivities
Yes    191318
No      54704
Name: count, dtype: int64

HadHeartAttack
No     232587
Yes     13435
Name: count, dtype: int64

HadAngina
No     231069
Yes     14953
Name: count, dtype: int64

HadStroke
No     235910
Yes     10112
Name: count, dtype: int64

HadAsthma
No     209493
Yes     36529
Name: count, dtype: int64

HadSkinCancer
No     225001
Yes     21021
Name: count, dtype: int64

HadCOPD
No     227028
Yes     18994
Name: count, dtype: int64

HadDepressiveDisorder
No     195402
Yes     50620
Name: count, dtype: int64

HadKidneyDisease
No     234738
Yes     11284
Name: count, dtype: int64

HadArthritis
No     161139
Yes     84883
Name: count, dtype: int64

DeafOrHardOfHearing
No     224990
Yes     21032
Name: count, dtype: int64

BlindOrVisionDifficulty
No     233796
Yes     12226
Name: count, dtype: int64

DifficultyConcentrating
No     219802
Yes     26220
Name: count, dtype: int64

DifficultyWalk

No outliers found in sex and binary-value String columns, all equal to 246022 values.

In [ ]:
# convert sex and binary-value String columns to binary integer values, for readability.

df['Sex'] = df['Sex'].replace(['Male','Female'], [0,1])
df['PhysicalActivities'] = df['PhysicalActivities'].replace(['Yes','No'], [1,0])
df['HadHeartAttack'] = df['HadHeartAttack'].replace(['Yes','No'], [1,0])
df['HadAngina'] = df['HadAngina'].replace(['Yes','No'], [1,0])
df['HadStroke'] = df['HadStroke'].replace(['Yes','No'], [1,0])
df['HadAsthma'] = df['HadAsthma'].replace(['Yes','No'], [1,0])
df['HadSkinCancer'] = df['HadSkinCancer'].replace(['Yes','No'], [1,0])
df['HadCOPD'] = df['HadCOPD'].replace(['Yes','No'], [1,0])
df['HadDepressiveDisorder'] = df['HadDepressiveDisorder'].replace(['Yes','No'], [1,0])
df['HadKidneyDisease'] = df['HadKidneyDisease'].replace(['Yes','No'], [1,0])
df['HadArthritis'] = df['HadArthritis'].replace(['Yes','No'], [1,0])
df['DeafOrHardOfHearing'] = df['DeafOrHardOfHearing'].replace(['Yes','No'], [1,0])
df['BlindOrVisionDifficulty'] = df['BlindOrVisionDifficulty'].replace(['Yes','No'], [1,0])
df['DifficultyConcentrating'] = df['DifficultyConcentrating'].replace(['Yes','No'], [1,0])
df['DifficultyWalking'] = df['DifficultyWalking'].replace(['Yes','No'], [1,0])
df['DifficultyDressingBathing'] = df['DifficultyDressingBathing'].replace(['Yes','No'], [1,0])
df['DifficultyErrands'] = df['DifficultyErrands'].replace(['Yes','No'], [1,0])
df['ChestScan'] = df['ChestScan'].replace(['Yes','No'], [1,0])
df['AlcoholDrinkers'] = df['AlcoholDrinkers'].replace(['Yes','No'], [1,0])
df['HIVTesting'] = df['HIVTesting'].replace(['Yes','No'], [1,0])
df['FluVaxLast12'] = df['FluVaxLast12'].replace(['Yes','No'], [1,0])
df['PneumoVaxEver'] = df['PneumoVaxEver'].replace(['Yes','No'], [1,0])
df['HighRiskLastYear'] = df['HighRiskLastYear'].replace(['Yes','No'], [1,0])

In [ ]:
df.head()

State  Sex GeneralHealth  PhysicalHealthDays  MentalHealthDays  \
0  Alabama    1     Very good                 4.0               0.0   
1  Alabama    0     Very good                 0.0               0.0   
2  Alabama    0     Very good                 0.0               0.0   
3  Alabama    1          Fair                 5.0               0.0   
4  Alabama    1          Good                 3.0              15.0   

                                     LastCheckupTime  PhysicalActivities  \
0  Within past year (anytime less than 12 months ...                   1   
1  Within past year (anytime less than 12 months ...                   1   
2  Within past year (anytime less than 12 months ...                   0   
3  Within past year (anytime less than 12 months ...                   1   
4  Within past year (anytime less than 12 months ...                   1   

   SleepHours            RemovedTeeth  HadHeartAttack  ...  HeightInMeters  \
0         9.0            None of them               0  ...            1.60   
1         6.0            None of them               0  ...            1.78   
2         8.0  6 or more, but not all               0  ...            1.85   
3         9.0            None of them               0  ...            1.70   
4         5.0                  1 to 5               0  ...            1.55   

   WeightInKilograms    BMI  AlcoholDrinkers  HIVTesting  FluVaxLast12  \
0              71.67  27.99                0           0             1   
1              95.25  30.13                0           0             1   
2             108.86  31.66                1           0             0   
3              90.72  31.32                0           0             1   
4              79.38  33.07                0           0             1   

   PneumoVaxEver                                  TetanusLast10Tdap  \
0              1                                 Yes, received Tdap   
1              1  Yes, received tetanus shot but not sure what type   
2              1  No, did not receive any tetanus shot in the pa...   
3              1  No, did not receive any tetanus shot in the pa...   
4              1  No, did not receive any tetanus shot in the pa...   

  HighRiskLastYear  CovidPos  
0                0        No  
1                0        No  
2                0       Yes  
3                0       Yes  
4                0        No  

[5 rows x 40 columns]

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 246022 entries, 0 to 246021
Data columns (total 40 columns):
 #   Column                     Non-Null Count   Dtype  
---  ------                     --------------   -----  
 0   State                      246022 non-null  object 
 1   Sex                        246022 non-null  int64  
 2   GeneralHealth              246022 non-null  object 
 3   PhysicalHealthDays         246022 non-null  float64
 4   MentalHealthDays           246022 non-null  float64
 5   LastCheckupTime            246022 non-null  object 
 6   PhysicalActivities         246022 non-null  int64  
 7   SleepHours                 246022 non-null  float64
 8   RemovedTeeth               246022 non-null  object 
 9   HadHeartAttack             246022 non-null  int64  
 10  HadAngina                  246022 non-null  int64  
 11  HadStroke                  246022 non-null  int64  
 12  HadAsthma                  246022 non-null  int64  
 13  HadSkinCancer              24

In [ ]:
# Check other values

df['GeneralHealth'].value_counts(dropna=False)

GeneralHealth
Very good    86999
Good         77409
Excellent    41525
Fair         30659
Poor          9430
Name: count, dtype: int64

In [ ]:
df['GeneralHealth'] = df['GeneralHealth'].replace(['Poor','Fair','Good','Very good','Excellent'], [0,1,2,3,4])
df['GeneralHealth'].value_counts(dropna=False)

GeneralHealth
3    86999
2    77409
4    41525
1    30659
0     9430
Name: count, dtype: int64

In [ ]:
df['LastCheckupTime'].value_counts(dropna=False)

LastCheckupTime
Within past year (anytime less than 12 months ago)         198153
Within past 2 years (1 year but less than 2 years ago)      23227
Within past 5 years (2 years but less than 5 years ago)     13744
5 or more years ago                                         10898
Name: count, dtype: int64

In [ ]:
df['LastCheckupTime'] = df['LastCheckupTime'].replace(['5 or more years ago','Within past 5 years (2 years but less than 5 years ago)','Within past 2 years (1 year but less than 2 years ago)','Within past year (anytime less than 12 months ago)'], [0,1,2,3])
df['LastCheckupTime'].value_counts(dropna=False)

LastCheckupTime
3    198153
2     23227
1     13744
0     10898
Name: count, dtype: int64

In [ ]:
df['RemovedTeeth'].value_counts(dropna=False)

RemovedTeeth
None of them              131592
1 to 5                     74702
6 or more, but not all     25950
All                        13778
Name: count, dtype: int64

In [ ]:
df['RemovedTeeth'] = df['RemovedTeeth'].replace(['None of them','1 to 5','6 or more, but not all','All'], [0,1,2,3])
df['RemovedTeeth'].value_counts(dropna=False)

RemovedTeeth
0    131592
1     74702
2     25950
3     13778
Name: count, dtype: int64

In [ ]:
df['HadDiabetes'].value_counts(dropna=False)

HadDiabetes
No                                         204834
Yes                                         33813
No, pre-diabetes or borderline diabetes      5392
Yes, but only during pregnancy (female)      1983
Name: count, dtype: int64

In [ ]:
df['HadDiabetes'] = df['HadDiabetes'].replace(['No','No, pre-diabetes or borderline diabetes','Yes, but only during pregnancy (female)','Yes'], [0,1,1,2])
df['HadDiabetes'].value_counts(dropna=False)

HadDiabetes
0    204834
2     33813
1      7375
Name: count, dtype: int64

In [ ]:
df['SmokerStatus'].value_counts(dropna=False)

SmokerStatus
Never smoked                             147737
Former smoker                             68527
Current smoker - now smokes every day     21659
Current smoker - now smokes some days      8099
Name: count, dtype: int64

In [ ]:
df['SmokerStatus'] = df['SmokerStatus'].replace(['Never smoked','Former smoker','Current smoker - now smokes some days','Current smoker - now smokes every day'], [0,1,2,3])
df['SmokerStatus'].value_counts(dropna=False)

SmokerStatus
0    147737
1     68527
3     21659
2      8099
Name: count, dtype: int64

In [ ]:
df['ECigaretteUsage'].value_counts(dropna=False)

ECigaretteUsage
Never used e-cigarettes in my entire life    190128
Not at all (right now)                        43281
Use them some days                             6658
Use them every day                             5955
Name: count, dtype: int64

In [ ]:
df['ECigaretteUsage'] = df['ECigaretteUsage'].replace(['Never used e-cigarettes in my entire life','Not at all (right now)','Use them some days','Use them every day'], [0,1,2,3])
df['ECigaretteUsage'].value_counts(dropna=False)

ECigaretteUsage
0    190128
1     43281
2      6658
3      5955
Name: count, dtype: int64

In [ ]:
df['AgeCategory'].value_counts(dropna=False)

AgeCategory
Age 65 to 69       28557
Age 60 to 64       26720
Age 70 to 74       25739
Age 55 to 59       22224
Age 50 to 54       19913
Age 75 to 79       18136
Age 80 or older    17816
Age 40 to 44       16973
Age 45 to 49       16753
Age 35 to 39       15614
Age 30 to 34       13346
Age 18 to 24       13122
Age 25 to 29       11109
Name: count, dtype: int64

In [ ]:
df['AgeCategory'] = df['AgeCategory'].replace(['Age 18 to 24',
                                               'Age 25 to 29',
                                               'Age 30 to 34',
                                               'Age 35 to 39',
                                               'Age 40 to 44',
                                               'Age 45 to 49',
                                               'Age 50 to 54',
                                               'Age 55 to 59',
                                               'Age 60 to 64',
                                               'Age 65 to 69',
                                               'Age 70 to 74',
                                               'Age 75 to 79',
                                               'Age 80 or older'], [0,0,1,1,2,3,4,5,6,7,8,9,10])
df['AgeCategory'].value_counts(dropna=False)

AgeCategory
1     28960
7     28557
6     26720
8     25739
0     24231
5     22224
4     19913
9     18136
10    17816
2     16973
3     16753
Name: count, dtype: int64

In [ ]:
df['TetanusLast10Tdap'].value_counts(dropna=False)

TetanusLast10Tdap
No, did not receive any tetanus shot in the past 10 years    81747
Yes, received tetanus shot but not sure what type            74119
Yes, received Tdap                                           70286
Yes, received tetanus shot, but not Tdap                     19870
Name: count, dtype: int64

In [ ]:
df['TetanusLast10Tdap'] = df['TetanusLast10Tdap'].replace(['No, did not receive any tetanus shot in the past 10 years',
                                                           'Yes, received tetanus shot, but not Tdap',
                                                           'Yes, received tetanus shot but not sure what type',
                                                           'Yes, received Tdap'], [0,1,2,3])
df['TetanusLast10Tdap'].value_counts(dropna=False)

TetanusLast10Tdap
0    81747
2    74119
3    70286
1    19870
Name: count, dtype: int64

In [ ]:
df['CovidPos'].value_counts(dropna=False)

CovidPos
No                                                               167306
Yes                                                               70324
Tested positive using home test without a health professional      8392
Name: count, dtype: int64

In [ ]:
df['CovidPos'] = df['CovidPos'].replace(['No','Yes','Tested positive using home test without a health professional'], [0,1,1])
df['CovidPos'].value_counts(dropna=False)

CovidPos
0    167306
1     78716
Name: count, dtype: int64

In [ ]:
# removal of category potentially leading to bias/discrimination
# or do not impact individual
df = df.drop(columns=['RaceEthnicityCategory', 'State'])

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 246022 entries, 0 to 246021
Data columns (total 38 columns):
 #   Column                     Non-Null Count   Dtype  
---  ------                     --------------   -----  
 0   Sex                        246022 non-null  int64  
 1   GeneralHealth              246022 non-null  int64  
 2   PhysicalHealthDays         246022 non-null  float64
 3   MentalHealthDays           246022 non-null  float64
 4   LastCheckupTime            246022 non-null  int64  
 5   PhysicalActivities         246022 non-null  int64  
 6   SleepHours                 246022 non-null  float64
 7   RemovedTeeth               246022 non-null  int64  
 8   HadHeartAttack             246022 non-null  int64  
 9   HadAngina                  246022 non-null  int64  
 10  HadStroke                  246022 non-null  int64  
 11  HadAsthma                  246022 non-null  int64  
 12  HadSkinCancer              246022 non-null  int64  
 13  HadCOPD                    24

In [ ]:
df

Sex  GeneralHealth  PhysicalHealthDays  MentalHealthDays  \
0         1              3                 4.0               0.0   
1         0              3                 0.0               0.0   
2         0              3                 0.0               0.0   
3         1              1                 5.0               0.0   
4         1              2                 3.0              15.0   
...     ...            ...                 ...               ...   
246017    0              3                 0.0               0.0   
246018    1              1                 0.0               7.0   
246019    0              2                 0.0              15.0   
246020    1              4                 2.0               2.0   
246021    0              3                 0.0               0.0   

        LastCheckupTime  PhysicalActivities  SleepHours  RemovedTeeth  \
0                     3                   1         9.0             0   
1                     3                   1         6.0             0   
2                     3                   0         8.0             2   
3                     3                   1         9.0             0   
4                     3                   1         5.0             1   
...                 ...                 ...         ...           ...   
246017                2                   1         6.0             0   
246018                3                   1         7.0             0   
246019                3                   1         7.0             1   
246020                3                   1         7.0             0   
246021                3                   0         5.0             0   

        HadHeartAttack  HadAngina  ...  HeightInMeters  WeightInKilograms  \
0                    0          0  ...            1.60              71.67   
1                    0          0  ...            1.78              95.25   
2                    0          0  ...            1.85             108.86   
3                    0          0  ...            1.70              90.72   
4                    0          0  ...            1.55              79.38   
...                ...        ...  ...             ...                ...   
246017               0          0  ...            1.78             102.06   
246018               0          0  ...            1.93              90.72   
246019               0          0  ...            1.68              83.91   
246020               0          0  ...            1.70              83.01   
246021               1          0  ...            1.83             108.86   

          BMI  AlcoholDrinkers  HIVTesting  FluVaxLast12  PneumoVaxEver  \
0       27.99                0           0             1              1   
1       30.13                0           0             1              1   
2       31.66                1           0             0              1   
3       31.32                0           0             1              1   
4       33.07                0           0             1              1   
...       ...              ...         ...           ...            ...   
246017  32.28                1           0             0              0   
246018  24.34                0           0             0              0   
246019  29.86                1           1             1              1   
246020  28.66                0           1             1              0   
246021  32.55                0           1             1              1   

        TetanusLast10Tdap  HighRiskLastYear  CovidPos  
0                       3                 0         0  
1                       2                 0         0  
2                       0                 0         1  
3                       0                 0         1  
4                       0                 0         0  
...                   ...               ...       ...  
246017                  2                 0         0  
246018                  0               

# REDUCING FEATURES

In [ ]:
# Combining Heart Attack and Heart Stroke
df['HadHeartAttackOrStroke'] = df['HadHeartAttack'] + df['HadStroke']
df['HadHeartAttackOrStroke'] = df['HadHeartAttackOrStroke'].where(df['HadHeartAttackOrStroke'] <= 1, other=1)
df['HadHeartAttackOrStroke'].value_counts(dropna=False)

HadHeartAttackOrStroke
0    224993
1     21029
Name: count, dtype: int64

In [ ]:
df.drop(columns=['HadHeartAttack', 'HadStroke'], inplace=True)
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 246022 entries, 0 to 246021
Data columns (total 37 columns):
 #   Column                     Non-Null Count   Dtype  
---  ------                     --------------   -----  
 0   Sex                        246022 non-null  int64  
 1   GeneralHealth              246022 non-null  int64  
 2   PhysicalHealthDays         246022 non-null  float64
 3   MentalHealthDays           246022 non-null  float64
 4   LastCheckupTime            246022 non-null  int64  
 5   PhysicalActivities         246022 non-null  int64  
 6   SleepHours                 246022 non-null  float64
 7   RemovedTeeth               246022 non-null  int64  
 8   HadAngina                  246022 non-null  int64  
 9   HadAsthma                  246022 non-null  int64  
 10  HadSkinCancer              246022 non-null  int64  
 11  HadCOPD                    246022 non-null  int64  
 12  HadDepressiveDisorder      246022 non-null  int64  
 13  HadKidneyDisease           24

In [ ]:
# Combining Physical Disabilities as Quantity
df['PhysicalDisabilities'] = df['DeafOrHardOfHearing'] + df['BlindOrVisionDifficulty'] + df['DifficultyConcentrating'] + df['DifficultyWalking'] + df['DifficultyDressingBathing'] + df['DifficultyErrands']
df['PhysicalDisabilities'].value_counts(dropna=False)

PhysicalDisabilities
0    174347
1     42575
2     16400
3      7649
4      3572
5      1212
6       267
Name: count, dtype: int64

In [ ]:
df = df.drop(columns=['DeafOrHardOfHearing', 'BlindOrVisionDifficulty', 'DifficultyConcentrating', 'DifficultyWalking', 'DifficultyDressingBathing', 'DifficultyErrands'])
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 246022 entries, 0 to 246021
Data columns (total 32 columns):
 #   Column                  Non-Null Count   Dtype  
---  ------                  --------------   -----  
 0   Sex                     246022 non-null  int64  
 1   GeneralHealth           246022 non-null  int64  
 2   PhysicalHealthDays      246022 non-null  float64
 3   MentalHealthDays        246022 non-null  float64
 4   LastCheckupTime         246022 non-null  int64  
 5   PhysicalActivities      246022 non-null  int64  
 6   SleepHours              246022 non-null  float64
 7   RemovedTeeth            246022 non-null  int64  
 8   HadAngina               246022 non-null  int64  
 9   HadAsthma               246022 non-null  int64  
 10  HadSkinCancer           246022 non-null  int64  
 11  HadCOPD                 246022 non-null  int64  
 12  HadDepressiveDisorder   246022 non-null  int64  
 13  HadKidneyDisease        246022 non-null  int64  
 14  HadArthritis        

In [ ]:
# Combine smoker status and ecig usage (weighted by smoker status having greater impact)
df['SmokerRating'] = df['SmokerStatus'] + df['ECigaretteUsage']/2
df['SmokerRating'].value_counts(dropna=False)

SmokerRating
0.0    126861
1.0     50647
0.5     17329
1.5     16141
3.0     11613
3.5      9164
2.0      6056
2.5      5880
4.0      1609
4.5       722
Name: count, dtype: int64

In [ ]:
df.drop(columns=['SmokerStatus', 'ECigaretteUsage'], inplace=True)

In [ ]:
# Remove redundant columns (weight and height vs BMI)
df = df.drop(columns=['WeightInKilograms','HeightInMeters'])


In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 246022 entries, 0 to 246021
Data columns (total 29 columns):
 #   Column                  Non-Null Count   Dtype  
---  ------                  --------------   -----  
 0   Sex                     246022 non-null  int64  
 1   GeneralHealth           246022 non-null  int64  
 2   PhysicalHealthDays      246022 non-null  float64
 3   MentalHealthDays        246022 non-null  float64
 4   LastCheckupTime         246022 non-null  int64  
 5   PhysicalActivities      246022 non-null  int64  
 6   SleepHours              246022 non-null  float64
 7   RemovedTeeth            246022 non-null  int64  
 8   HadAngina               246022 non-null  int64  
 9   HadAsthma               246022 non-null  int64  
 10  HadSkinCancer           246022 non-null  int64  
 11  HadCOPD                 246022 non-null  int64  
 12  HadDepressiveDisorder   246022 non-null  int64  
 13  HadKidneyDisease        246022 non-null  int64  
 14  HadArthritis        

In [ ]:
# Combine lung diseases
df['HadLungDisease'] = df['HadAsthma'] + df['HadCOPD']
df['HadLungDisease'] = df['HadLungDisease'].where(df['HadLungDisease'] <= 1, other=1)
df.drop(columns=['HadAsthma', 'HadCOPD'], inplace=True)

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 246022 entries, 0 to 246021
Data columns (total 28 columns):
 #   Column                  Non-Null Count   Dtype  
---  ------                  --------------   -----  
 0   Sex                     246022 non-null  int64  
 1   GeneralHealth           246022 non-null  int64  
 2   PhysicalHealthDays      246022 non-null  float64
 3   MentalHealthDays        246022 non-null  float64
 4   LastCheckupTime         246022 non-null  int64  
 5   PhysicalActivities      246022 non-null  int64  
 6   SleepHours              246022 non-null  float64
 7   RemovedTeeth            246022 non-null  int64  
 8   HadAngina               246022 non-null  int64  
 9   HadSkinCancer           246022 non-null  int64  
 10  HadDepressiveDisorder   246022 non-null  int64  
 11  HadKidneyDisease        246022 non-null  int64  
 12  HadArthritis            246022 non-null  int64  
 13  HadDiabetes             246022 non-null  int64  
 14  ChestScan           